File: Baekjoon/1260.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [info, ...edges] = fs.readFileSync(filePath).toString().trim().split("\n");
let [n, m, v] = info.split(" ");
let graph = new Array(+n + 1).fill().map(() => []);
let visited = new Array(+n + 1).fill(0);
let dfsArr = [];
let bfsArr = [];

const dfs = (v) => {
  if (visited[v] == true) return;

  visited[v] = 1;
  dfsArr.push(+v);

  for (let i = 0; i < graph[v].length; i++) {
    let next = graph[v][i];
    if (visited[next] == 0) dfs(next);
  }
};

const bfs = (v) => {
  let queue = [+v];

  while (queue.length > 0) {
    let cur = queue.shift();

    if (visited[cur] == 1) continue;
    visited[cur] = 1;
    bfsArr.push(cur);

    for (let i = 0; i < graph[cur].length; i++) {
      let next = graph[cur][i];
      if (visited[next] == 0) queue.push(next);
    }
  }
};

const result = (v) => {
  for (let i = 0; i < +m; i++) {
    let [from, to] = edges[i].split(" ");
    graph[from].push(+to);
    graph[to].push(+from);
  }
  graph.forEach((ele) => {
    ele.sort((a, b) => a - b);
  });

  dfs(v);
  console.log(dfsArr.join(" "));

  visited.fill(0);

  bfs(v);
  console.log(bfsArr.join(" "));
};

result(v);

File: python/Baekjoon/1260.py
from collections import deque

n, m, v = map(int, input().split())
graph = [[] for i in range(n+1)]
for i in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
for j in graph:
    j.sort()
visited = [False] * (n + 1)
dfsResult = []
bfsResult = []

def dfs(graph, v, visited):
    visited[v] = True
    dfsResult.append(v)
    for i in graph[v]:
        if not visited[i]: dfs(graph, i, visited)

def bfs(graph, v, visited):
    queue = deque()
    queue.append(v)
    visited[v] = True
    while queue:
        cur = queue.popleft()
        bfsResult.append(cur)
        for i in graph[cur]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True

dfs(graph, v, visited)
visited = [False] * (n + 1)
bfs(graph, v, visited)

print(" ".join(map(str, dfsResult)))
print(" ".join(map(str, bfsResult)))
